ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/MetaStoreCompactorThread.java

```diff
@@ -128,9 +132,40 @@ protected static long updateCycleDurationMetric(String metric, long startedAt) {
       long elapsed = System.currentTimeMillis() - startedAt;
       LOG.debug("Updating {} metric to {}", metric, elapsed);
       Metrics.getOrCreateGauge(metric)
-          .set((int)elapsed);
+          .set((int) elapsed);
       return elapsed;
     }
     return 0;
   }
+
+  @VisibleForTesting
+  protected static void waitAllAsyncTask(List<CompletableFuture<Void>> tasks) throws AsyncTaskCompletionException {
+    List<Throwable> exceptions = new ArrayList<>();
+    for (CompletableFuture<Void> task : tasks) {
+      try {
+        task.join();
```

Comment:
isn't it the same as CompletableFuture.allOf(List<CompletableFuture> cf).join() ? you could do exceptions++ when declaring CompletableFuture by adding 
````
.exceptionally(exception -> {
    collectedExceptions.add(exception);
    return null;
}
````

Reply:
Which thread safe list implementation do you prefer in this case?

Reply:
Or simply an atomic integer would suffice